Alpaca/New-AlpacaBcContainer.ps1

```diff
@@ -0,0 +1,146 @@
+    $containerId = New-AlpacaBcContainer `
+        -authContext $authContext `
+        -artifactUrl $artifactUrl `
+        -credential $credential `
+        -containerName 'bcserver' `
+        -doNotWait `
+        -devContainer
+    Wait-AlpacaBcContainerReady `
+        -authContext $authContext `
+        -containerId $containerId
+#>
+function New-AlpacaBcContainer {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [Alias('AlpacaAuthContext')]
+        [HashTable] $authContext,
+        [Parameter(Mandatory=$true)]
+        [string] $artifactUrl,
+        [string] $licenseFileUrl = '',
+        [switch] $devContainer,
+        [string] $containerName = $bcContainerHelperConfig.defaultContainerName,
+        [string] $sourceLabel = 'BcContainerHelper',
+        [string[]] $environmentVariables = @(),
+        [hashTable] $additionalLabels = @{},
+        [string[]] $additionalFolders = @(),
+        [PSCredential] $credential,
+        [switch] $enableSymbolLoading,
+        [switch] $enablePremium,
+        [switch] $enablePerformanceCounter,
+        [switch] $publishSwaggerUI,
+        [string] $autoStartTime = '08:00',
+        [string] $autoStartTimeUTC = [datetime]::ParseExact($autoStartTime,'H:mm',[CultureInfo]::InvariantCulture).ToUniversalTime().ToString('HH:mm'),
+        [ValidateSet('deactivated','daily','weekdays','once')]
+        [string] $autoStartMode = "$(if($devContainer){"weekdays"}else{"deactivated"})",
+        [string] $autoStopTime = '18:00',
+        [string] $autoStopTimeUTC = [datetime]::ParseExact($autoStopTime,'H:mm',[CultureInfo]::InvariantCulture).ToUniversalTime().ToString('HH:mm'),
+        [ValidateSet('daily','deactivated','deactivated once')]
+        [string] $autoStopMode = "$(if($devContainer){"daily"}else{"deactivated"})",
+        [switch] $doNotWait
+    )
+
+    $uri, $headers = Get-UriAndHeadersForAlpaca -authContext $authContext -serviceUrl "Service/bc"
+
+    # Set default AZP_SERVICE_DISPLAYNAME (can be overridden in $environmentVariables)
+    $environmentVariables = @( "AZP_SERVICE_DISPLAYNAME=$containerName" ) + $environmentVariables
+    $additionalLabels.Source = $sourceLabel
+    # Set Alpaca additional Folder for c:\run first - can be overridden in $additionalFolders
+    if ($devContainer) {
+        $typeStr = 'dev'
+    }
+    else {
+        $typeStr = 'build'
+    }
+    $runFolderUrl = [string]::Format($bcContainerHelperConfig.AlpacaSettings.RunFolderUrl,$typeStr,$bcContainerHelperConfig.AlpacaSettings.ApiVersion)
+    $additionalFolders = @( "c:\run=$runFolderUrl" ) + $additionalFolders
+
+    $parameters = [ordered]@{
+        "persistence" = $devContainer.IsPresent
+        "environmentVars" = $environmentVariables
+        "additionalLabels" = $additionalLabels
+        "additionalFolders" = $additionalFolders
+        "licenseFile" = $licenseFileUrl
+        "bcArtifact" = @{
+            "storageAccount" = $artifactUrl.Split('/')[2].Split('.')[0]

```

Comment:
We have an "intelligent image selection" in the service that prefers images that are already available so that image builds happen very rarely. We instead rebuild images every night so that all new images are available for the next day and only rebuild during the day if the user requests a specific version that isn't available yet.

To make this "intelligent image selection" work best, we would instead need the requested version instead of the full version. Fo example if the user requests BC 20, container will be created with newest BC 20 version that is already available. Every night the newest image will be rebuilt so the user might get new images delayed by 1 day.

So the best would be if `bcArtifact` has the same values that a user would pass into `get-bcartifacturl`

Reply:
OK, I have changed this to support the format of the artifact parameter in AL-Go.
Either the full ArtifactUrl if they indeed need that specific artifact
or a string using storageAccount/type/version/country (where version can be like 21.5)

Reply:
Understood. But how could I specify select=nextmajor this way?

Reply:
I actually have a select as well storageAccount/type/version/country/select
How should I do this with Alpaca?

Reply:
meaning that ///us/nextmajor would give me nextmajor (which corresponds to giving these parameters to get-bcartifactUrl)

Reply:
there's another field `select` in `bcArtifact` that then can be set to the according value

Reply:
and what about the Insider sas token - is that also a field then?

Reply:
that is set up in the backend per-tenant. Btw I remember that you thought about removing insider sas token requirement completely - what's the state on this? would make onboarding even easier